AT-3/src/at3/Vegetable.java

```diff
@@ -0,0 +1,88 @@
+package at3;
+
+import java.util.Comparator;
+
+public class Vegetable {
+    private String name;
+    private String color;
+    private double weight;
+    private double energyDensity; //kcal/100g
+
+    Vegetable(String name, String color, double weight, double energyDensity) {
+        this.name = name;
+        this.color = color;
+        this.weight = weight;
+        this.energyDensity = energyDensity;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public double getWeight() {
+        return weight;
+    }
+
+    public double getEnergyDensity() {
+        return energyDensity;
+    }
+
+    public double calcCalories() {
+        double result = (energyDensity * weight) / 100;
+        return result;
+    }
+    public boolean equals(Vegetable v) {

```

Comment:
you need use @Override tag to override standard java implementation from Object class